### PR TITLE
[GLUTEN-4695][VL] Fix array type Substrait signature

### PR DESCRIPTION
**Pull request by liujiayi771**
## What changes were proposed in this pull request?

We need to pass elementType in ArrayType to Substrait signature.

## How was this patch tested?

Add test case in `TestOperator`.

### CODE REVIEW

**Comment by liujiayi771** [comment]
velox不支持array比较时其中包含null，先exclude。
```
2024-02-12T03:36:26.3508665Z Caused by: java.lang.RuntimeException: Exception: VeloxUserError
2024-02-12T03:36:26.3509365Z Error Source: USER
2024-02-12T03:36:26.3509736Z Error Code: INVALID_ARGUMENT
2024-02-12T03:36:26.3510401Z Reason: ARRAY comparison not supported for values that contain nulls
2024-02-12T03:36:26.3511114Z Retriable: False
2024-02-12T03:36:26.3511767Z Expression: !decoded.base()->containsNullAt(indices[index])
2024-02-12T03:36:26.3512426Z Function: checkNestedNulls
2024-02-12T03:36:26.3512982Z File: ../../velox/functions/lib/CheckNestedNulls.cpp
2024-02-12T03:36:26.3513643Z Line: 34
2024-02-12T03:36:26.3513933Z Stack trace:
2024-02-12T03:36:26.3516454Z # 0  facebook::velox::VeloxException::VeloxException(char const*, unsigned long, char const*, std::basic_string_view<char, std::char_traits<char> >, std::basic_string_view<char, std::char_traits<char> >, std::basic_string_view<char, std::char_traits<char> >, std::basic_string_view<char, std::char_traits<char> >, bool, facebook::velox::VeloxException::Type, std::basic_string_view<char, std::char_traits<char> >)
2024-02-12T03:36:26.3520742Z # 1  void facebook::velox::detail::veloxCheckFail<facebook::velox::VeloxUserError, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&>(facebook::velox::detail::VeloxCheckFailArgs const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)
2024-02-12T03:36:26.3523237Z # 2  facebook::velox::functions::checkNestedNulls(facebook::velox::DecodedVector const&, int const*, int, bool)
2024-02-12T03:36:26.3525717Z # 3  facebook::velox::aggregate::prestosql::(anonymous namespace)::NonNumericMinAggregate::addRawInput(char**, facebook::velox::SelectivityVector const&, std::vector<std::shared_ptr<facebook::velox::BaseVector>, std::allocator<std::shared_ptr<facebook::velox::BaseVector> > > const&, bool)
2024-02-12T03:36:26.3527961Z # 4  facebook::velox::exec::StreamingAggregation::evaluateAggregates()
2024-02-12T03:36:26.3528861Z # 5  facebook::velox::exec::StreamingAggregation::getOutput()
2024-02-12T03:36:26.3530397Z # 6  facebook::velox::exec::Driver::runInternal(std::shared_ptr<facebook::velox::exec::Driver>&, std::shared_ptr<facebook::velox::exec::BlockingState>&, std::shared_ptr<facebook::velox::RowVector>&)
2024-02-12T03:36:26.3532127Z # 7  facebook::velox::exec::Driver::next(std::shared_ptr<facebook::velox::exec::BlockingState>&)
2024-02-12T03:36:26.3533129Z # 8  facebook::velox::exec::Task::next(folly::SemiFuture<folly::Unit>*)
2024-02-12T03:36:26.3533861Z # 9  gluten::WholeStageResultIterator::next()
2024-02-12T03:36:26.3534623Z # 10 Java_io_glutenproject_vectorized_ColumnarBatchOutIterator_nativeHasNext
2024-02-12T03:36:26.3535336Z # 11 0x00007f189e7bfc00
```

**Comment by liujiayi771** [comment]
@PHILO-HE @rui-mo Could you help to review?

File: cpp/velox/substrait/VeloxSubstraitSignature.cc [hunk 1]
```diff
@@ -168,6 +168,20 @@ TypePtr VeloxSubstraitSignature::fromSubstraitSignature(const std::string& signa
     }
     return std::make_shared<RowType>(std::move(names), std::move(types));
   }
+
+  if (startWith(signature, "list")) {
+    auto listStart = signature.find_first_of('<');
+    auto listEnd = signature.find_first_of('>');
```
Comment (PHILO-HE) [inline cpp/velox/substrait/VeloxSubstraitSignature.cc:174]: Do we need to consider nested array? For nested array, `find_first_of('>')` will return wrong ending position for outer array, right?

**Review by PHILO-HE — COMMENTED** [review]
Thanks for your work! Could you please create an issue to record array containing null is not supported?

File: cpp/velox/substrait/VeloxSubstraitSignature.cc (hunk 1 above)
Comment (liujiayi771) [reply cpp/velox/substrait/VeloxSubstraitSignature.cc:174]: Yes, we should use `find_last_of`.

**Comment by liujiayi771** [comment]
@PHILO-HE Create an issue https://github.com/oap-project/gluten/issues/4695.

**Review by PHILO-HE — APPROVED** [review]
Looks good! Thanks!

**Comment by PHILO-HE** [comment]
Gluten ClickHouse CI is red. Please check it.

**Comment by liujiayi771** [comment]
> Gluten ClickHouse CI is red. Please check it.

CH CI has passed.

**Pull request merged**